packages/xo-web/src/xo-app/user/index.js

```diff
@@ -76,6 +79,81 @@ const getUserPreferences = user => user.preferences || {}
 
 // ===================================================================
 
+@addSubscriptions({
+  user: subscribeCurrentUser,
+})
+class XsClientId extends Component {
+  async editXsCredentials(file) {
+    if (file === undefined) {
+      error(_('noFileSelected'))
+      return
+    }
+
+    try {
+      await new Promise((resolve, reject) => {
+        const fr = new window.FileReader()
+        fr.onload = event => {
+          try {
+            const { username, apikey } = JSON.parse(event.target.result)
+            if (username === undefined || apikey === undefined) {
+              reject(new Error('Could not find username and apikey in file'))
+            }
+
+            editXsCredentials({ username, password: apikey }).then(resolve, reject)
```

Comment:
Why not keep the `apiKey` name?

IMHO renaming to `password` only introduces confusion in the rest of the code.

Reply:
Because it's only used for HTTP authentication when we download updates and it's usually called username and password during the authentication. But I don't have a strong opinion about it so I can change it.

Reply:
Yep, that's much better IMO, thanks.